polars/polars-time/src/date_range.rs

```diff
@@ -1,14 +1,37 @@
-use chrono::{Datelike, NaiveDateTime};
+#[cfg(feature = "timezones")]
+use arrow::temporal_conversions::{
+    parse_offset, timestamp_ms_to_datetime, timestamp_ns_to_datetime, timestamp_us_to_datetime,
+};
+#[cfg(feature = "timezones")]
+use chrono::TimeZone as TimeZoneTrait;
+use chrono::{Datelike, FixedOffset, NaiveDateTime};
 use polars_core::prelude::*;
 use polars_core::series::IsSorted;
 
 use crate::prelude::*;
+#[cfg(feature = "timezones")]
+use crate::windows::duration::localize_datetime;
 
 pub fn in_nanoseconds_window(ndt: &NaiveDateTime) -> bool {
     // ~584 year around 1970
     !(ndt.year() > 2554 || ndt.year() < 1386)
 }
 
+#[cfg(feature = "timezones")]
+fn localize_timestamp<T: TimeZoneTrait>(timestamp: i64, tu: TimeUnit, tz: T) -> PolarsResult<i64> {
```

Comment:
I don't think we will use this in a tight loop. If we do, we can factor out the duplicate branching and &fn indirection:

```rust
#[cfg(feature = "timezones")]
fn localize_timestamp<T: TimeZoneTrait>(timestamp: i64, tu: TimeUnit, tz: T) -> PolarsResult<i64> {
    Ok(match tu {
        TimeUnit::Nanoseconds => localize_datetime(timestamp_ns_to_datetime(timestamp), &tz)?.timestamp_nanos(),
        TimeUnit::Microseconds => localize_datetime(timestamp_us_to_datetime(timestamp), &tz)?.timestamp_micros(),
        TimeUnit::Milliseconds => localize_datetime(timestamp_ms_to_datetime(timestamp), &tz)?.timestamp_millis(),
    })
}
```

I expect that to be faster, but for now it is fine. :+1:

Reply:
it's only used once upon function entry - when were you looking to release? I have about 20 mins before starting work, I'll try to address